src/coreclr/jit/fgprofilesynthesis.cpp

```diff
@@ -252,6 +338,8 @@ void ProfileSynthesis::AssignLikelihoods()
 
             case BBJ_COND:
                 // Two successor cases
+                block->GetTrueEdge()->setHeuristicBased(true);
+                block->GetFalseEdge()->setHeuristicBased(true);
```

Comment:
`ProfileSynthesis::AssignLikelihoods` is currently the only place where we set the heuristic-derived flag. There are plenty of sites in other phases where we set likelihoods based on some heuristic (usually next to a TODO suggesting we verify the likelihood makes sense) -- I should probably spot-check these and set the flag there as well. I've yet to see them cause issues with throw likelihoods, so I've left that out of this PR to reduce clutter.